### ISSUE TITLE
Display comments directly

### ISSUE THREAD

**Issue by lpointet** [uncatcrea/wp-appkit#216]
Thanks to deep linking, we can now open the app on any screen. This means we can choose to display some comments (e.g. with route `#comments-42`).
Internally, comments are bound to the fact that a post should have been displayed before them (ID 42 in the example), and some things break if we try to bypass this logic:
* on Android, TemplateTags.getPostLink( post.id ) doesn't return anything because of previous screen's `global` value not being set
* on iOS, it seems a JS error is raised when trying to directly go to such a route: we only see the top of the page

**Comment by mleroi** [comment]
Thanks @lpointet for creating the issue. 
I take that, I think I will "manually" add the post to history if it's not already in it when we display a comment screen.

**Comment by mleroi** [comment]
I handled the case for comments screen that correspond to a **post** that is in the app: we search for the corresponding post (in local storage) and add its corresponding screen to history manually before inserting the comment screen. 

The case where the comments screen corresponds to a **page** is more complicated, because pages are linked to components and we can't know which component the page belongs to, so we can't really create the page screen to add it to history...
It makes me think that maybe we should allow to display a page outside its component context, but I'll have to think a bit more about that.

**Comment by lpointet** [comment]
@mleroi Pages can actually be displayed outside of page component, if you choose a "Post List" component with "Pages" post type. :)
You will then have pages shown through routes like /single/posts/ID

**Comment by mleroi** [comment]
Yes, and actually you can also display a page (from page component) with route like #single/pages/ID.
But those #single/* routes will render using the "single.html" template instead of "page.html", which is not what we want in most cases as page.html is often different from single.html.

What happens for now is:
- trying to access #comments-[post-ID] directly
- we check if post-ID is in app's posts (if it is, we're ok, we add the post screen to history, that works now)
- if not, we check if post-ID is in app's pages
- if it is, we should add the page screen to history, then add the comment screen
But we can't know what component the page belongs to only from the page ID, so we can't create the page screen object properly :(

The reason why we need to know the component a page belongs to when displaying the page is because the page's "tree properties" depend on the page's component (for example a same page can be the root of one component, and be part of the subtree in another component) so its depth and its relation to other pages will be different according to the component it belongs to...

This means in fact that knowing the page's component is important only for "page tree" components. Single page components do not really need to know their component to display properly.

So I'm considering adding a new route type **#page/[page-ID]** (in addition to current page route *#page/[component-ID]/[page-ID]*) that would display a page using "page.html" and considering the page as "not being a tree page" (even if it is). This would be quite convenient in many cases where we just want to display a single page without caring about its component (ie its position in page tree).

Considering our deeplink problem with "access page comments screen directly" , we could also imagine adding a new route **#page/[component-ID]/[page-ID]/comments** that would be the solution to add the page screen to history in a robust way.

Maybe we can discuss all that in a next conf call :)

**Comment by uncatcrea** [comment]
New page routes will be addressed in #218
